### ISSUE TITLE
Feature request: Add totals/margins to flat_table

### ISSUE THREAD

**Issue by dannyparsons** [strengejacke/sjmisc#59]
I really like the functionality of `flat_table`. 

I was wondering if you would consider extending the functionality to be able to add totals as extra rows/columns to the table? Something similar to what `xtab` does or [addmargins](https://stat.ethz.ch/R-manual/R-devel/library/stats/html/addmargins.html) (not necessarily with the same flexibility)

I can sort of do it already:
```
efc %>%
    select(e16sex, e42dep, c172code, n4pstu, c161sex) %>%
    flat_table() %>% 
    addmargins()
```

although this loses the value labels.

Then if I add a `group_by` it doesn't work at all:
```
efc %>%
    group_by(e16sex, e42dep) %>%
    select(e16sex, e42dep, c172code, n4pstu, c161sex) %>%
    flat_table() %>% 
    addmargins()
# Error in addmargins(.) : 'A' must be an array or table
```
presumably because it's now multiple tables.

If it wouldn't be very difficult, I think it would be a really nice addition to `flat_table`.

**Comment by strengejacke** [comment]
> although this loses the value labels.

Yes, this was the reason why I haven't implemented this feature yet, because I haven't found a proper and easy solution that preserves the labels. But I'll see how to address this issue.

**Comment by dannyparsons** [comment]
Fair enough, thanks for the response. Look forward to seeing the progress.

**Comment by strengejacke** [comment]
I think I'll re-write the function from scratch. Will probably be the next greater change to _sjmisc_.
